### ISSUE TITLE
[bench request]: JPT-4B — Qwen3.5-4B LoRA, one-prefill option readout, TypeSafe wire format via llm2jev (self-hosted)

### ISSUE THREAD

**Issue by tic-top** [fstandhartinger/jevbench#78]
**Weights:** https://huggingface.co/kirp/jpt-4b at [`13f4884`](https://huggingface.co/kirp/jpt-4b/tree/13f4884984b252cfac5c8a675388f31fb5f54ec1) · **serving:** [llm2jev](https://pypi.org/project/llm2jev/) 0.6.1 (MIT, https://github.com/tic-top/llm2jev) in front of SGLang 0.5.9

JPT-4B is `Qwen/Qwen3.5-4B` with a LoRA (r=16, every attention / DeltaNet / MLP projection of the language model)
merged into full weights. The vision tower is untouched. It was trained with a multi-class Brier loss over the option
labels, on llm2jev's chat prompt with thinking disabled. That is the same prompt it is served with. Each question is
one prefill: the answer is read from the next-token distribution over the option labels, and no tokens are generated.

**What we measured with JevBench's CLI at `2fa63fa` on the public set (231 items):**

| | one A100 40 GB, SGLang 0.5.9 |
|---|---|
| correct | 203/231 |
| easy / standard / hard | 48/48 · 68/72 · 87/111 |
| answered and valid | 231/231 |
| ECE (hard) | 0.049 |
| latency p50 / p95 | 74 ms / 159 ms |
| mean input tokens | 668 |

**Identity check:** a correct setup reproduces 203/231 with that tier split.

**To run it.** llm2jev serves your wire format, so the existing `typesafe` adapter reaches it with no new code:

```
pip install "sglang==0.5.9" "llm2jev>=0.6.1"
pip install "nvidia-cudnn-cu12>=9.15"   # sglang 0.5.9's torch 2.9.1 pins cuDNN 9.10, which sglang refuses to run on
python -m sglang.launch_server --model-path kirp/jpt-4b --revision 13f4884984b252cfac5c8a675388f31fb5f54ec1 \
    --port 30000 --context-length 32768 --mamba-scheduler-strategy extra_buffer &
llm2jev --model kirp/jpt-4b --backend sglang --url http://127.0.0.1:30000 --port 8000 --temperature 1.036
python3 -m jevbench.cli run \
    --tasks datasets/public/easy.jsonl,datasets/public/original.jsonl,datasets/public/hard.jsonl \
    --adapter typesafe --endpoint http://127.0.0.1:8000 --model kirp/jpt-4b \
    --key-env "" --results out.jsonl
```

The first start downloads 9 GB. There is no hosted endpoint and none is planned, so the run is self-hosted on your
pods and the row needs no API flag.

**Disclosures**

- **Benchmark feedback:** we looked at our earlier checkpoints' public hard-tier misses by family, and that decided
  which kinds of training data we generated next: long policy documents with amendments, and constrained trade-offs.
  No JevBench item, and no text derived from one, was used for training, calibration or prompt selection. The
  generated documents were written by an LLM that was never shown a JevBench item. We dropped every document that
  shares more than three 8-word spans with the public items. The public set is still development evidence for us, not
  held-out evidence. The sealed items are the real test.
- **Calibration:** one global temperature (T = 1.036), fitted on a held-out split of our own training mix. Nothing is
  fitted per benchmark or per question type.
- **Context:** SGLang is started with `--context-length 32768`. The longest public item is about 4k tokens.
- **Option order:** options are presented as labelled lines in the order `criteria` lists them. Reversing the options
  of the 139 public `choice` questions changes 8 verdicts, and accuracy is 0.892 either way.
- `score` levels are shown in their given order as `0…n`. `noul` is shown as Yes/No.
- Structured state is flattened by llm2jev to indented text, not to JSON.
- **Price basis:** self-hosted bf16 on one A100 40 GB, input tokens only. Nothing is generated.
- **Speed:** latencies are from the runner as our harness invokes it on one A100 40 GB with no other load. We did not
  measure single-request latency separately.
- **Answers:** they carry `usage` (input / output tokens) next to the fields your adapter reads.

**Row details**

- **repo:** https://huggingface.co/kirp/jpt-4b
- **author:** kirp
- **underlying:** `Qwen/Qwen3.5-4B` (Alibaba Cloud), LoRA fine-tuned and merged.
- **licence:** CC BY-NC 4.0 for the weights. Some training datasets are non-commercial or research-only. llm2jev is
  MIT.

The sealed items are yours to run. We have never seen them and nothing is fitted to them.

**Comment by fstandhartinger** [comment]
Thanks — this is logged for reachability checks and the JevBench measurement queue. We’ll only make benchmark claims after our own run.

- Harold
